### ISSUE TITLE
Exception Value:	 super(type, obj): obj must be an instance or subtype of type

### ISSUE THREAD

**Issue by khrizo** [jrief/djangocms-cascade#51]
v: 0.4.1
When saving a new collapsable panel. 
cmsplugin_cascade/bootstrap3/collapse.py in get_identifier, line 64

Line 64: identifier = super(PanelGroupPlugin, cls).get_identifier(obj)

    @classmethod
    def get_identifier(cls, obj):
        identifier = super(PanelGroupPlugin, cls).get_identifier(obj)
        content = obj.glossary.get('panel_title', '')
        if content:
            content = unicode(Truncator(content).words(3, truncate=' ...'))
        return format_html('{0}{1}', identifier, content)

**Comment by khrizo** [comment]
I think this is related to something else, I tried a brand new installation and I can't replicate the issue.

Thanks for this great app.

**Comment by abhishekreddy1206** [comment]
i am getting this issue with django version 1.7.3

**Comment by lucaippo** [comment]
Same problem here with python 2.7, django 1.7.5, djangocms-cascade 0.4.2 and django-cms 3.0.12.
It looks like "obj" is of type "Exception":
![issue](https://cloud.githubusercontent.com/assets/2845439/6572076/4be1235c-c70f-11e4-9eac-1e8cbdaf6eb5.png)
It looks weird, maybe I'm missing something...

**Comment by jrief** [comment]
Can you built this reproducible?
Please fork the supplied demo app and specify the steps to reproduce this. Otherwise its impossible for me to find out what's going on.

**Comment by lucaippo** [comment]
I was able to reproduce this using bs3demo app against current master (which is the same of version 0.4.3) using Python 2.7.9:
- forked repo and cloned it in my local PC
- changed directory inside git checkout
- created new virtualenv
- installed requirements (`pip install -r bs3demo/requirements.txt`), it installed Django 1.7.6
- created database (`manage.py migrate --settings=bs3demo.settings`)
- start dev server (`manage.py runserver --settings=bs3demo.settings`)
- created superuser (`python manage.py createsuperuser --settings=bs3demo.settings`) to edit pages
- navigated to localhost:8000, it showed Django-CMS blue splashscreen
- clicked link for adding pages, and created HomePage
- after clicking "save" I got `TemplateDoesNotExist` for `bootstrap/menu/navbar.html`
- checked that the template was missing from local venv (it was missing), and in fact the djangocms-bootstrap3 package lacks the whole "templates" folder, both from pypi and from https://github.com/jrief/djangocms-bootstrap3/archive/master.zip
- downloaded .zip from github and placed cms_bootstrap3/templates content inside bs3demo/templates
- went back to localhost:8000: the page was there, but styles were missing
- used bower to install bootstrap/jquery as staed in documentation
- copied content of bower_components inside bs3demo/static
- went back to localhost:8000, and home page was fine
- switched to edit mode -> structure
- added (not fluid, default options) container to the only placeholder
- added row (1 column) to the container
- added panel group to the only column with 3 panels (when I first reported the issue, this is when I got the exception, and I was at work on a Windows 7 machine, while I was on Arch Linux when used your demo)
- the "structure" interface was saying that the panel group had 3 panels, but none of the were displayed: panel group was "blank", despite its caption
- tried to add another panel to panel group: it asked for a title, and I gave it one and saved
- structure interface said that panel group had 4 panels, while only one was there, but without the title i gave it (it said `<Empty>`)
- "content" mode showed all 4 panels, tiled up vertically, with some space in between them
- tried to add a text plugin inside the only displayed panel, it worked
- double-clicked the panel to edit title again, and saved 
- I got the exception
![screenshot1](https://cloud.githubusercontent.com/assets/2845439/6683628/181e14ba-cc89-11e4-8d34-90e47402545a.png)
![screenshot2](https://cloud.githubusercontent.com/assets/2845439/6683630/1b1fe09e-cc89-11e4-9e2c-1e4fc9c4c18c.png)
- checking database it looked that all 4 panel plugins were there (4 cms_cmsplugin instances, and 4 cmsplugin_cascade_element instances)
- trying to edit panel-group number of panels worked only if raising it (4 to 5), but didn't when lowering from 4 to 3 (it think this might be intended, not sure)
- trying to delete the only displayed panel (the one I added manually) worked
- trying to delete panel group raised exception

So I was able to reproduce the issue but in a bit different way this time: i had to edit a panel instead of just adding a panel group. Not sure why to be honest, and I still don't know why the exception is raised.

Hope it helps in some way, and if these steps won't help you understand what is going wrong, I'll try to dig a bit further in the code, since I'm still not very familiar with it, end eventually send a PR if I manage to find a solution.

Luca

**Comment by jrief** [comment]
Thanks for this fantastic detailed report.
Just another question, did this only happen with the PanelGroupPlugin or even with other plugin types?

About your report: “... but didn't when lowering from 4 to 3 ...”. Yes this is intentional behavior; one must explicitly chose the panel to delete.

I now will try to reproduce according to your description.

**Comment by lucaippo** [comment]
So far I've encountered this issue only with PanelGroupPlugin/PanelPlugin (not sure which of the 2 is causing troubles), and I think I've used most of the plugins you provide with this great app. I'll let you know if it happens with any other, ofc.

P.S.: I forgot to mention that the issue happens both on PostgreSQL (9.3) which I was using at work and with SQLite I used for this report, just to point out that I don't think it's something related to SQLite.
